internal/envoy/v3/route.go

```diff
@@ -71,6 +71,15 @@ func VirtualHostAndRoutes(vh *dag.VirtualHost, dagRoutes []*dag.Route, secure bo
 		evh.RateLimits = GlobalRateLimits(vh.RateLimitPolicy.Global.Descriptors)
 	}
 
+	if vh.RequestHeadersPolicy != nil {
```

Comment:
Please add some test cases for this new logic to `route_test.go`. Looks like we don't have an existing test for this function but now's a great time to start.

Reply:
Added